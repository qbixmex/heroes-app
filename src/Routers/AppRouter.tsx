import { BrowserRouter, Route, Routes } from "react-router-dom"
import DCScreen from '../Components/DC/DCScreen';
import LoginScreen from "../Components/Login/LoginScreen";
import MarvelScreen from '../Components/Marvel/MarvelScreen';
import SearchScreen from "../Components/Search/SearchScreen";
import { Navbar } from '../Components/UI/Navbar';

const AppRouter = () => {
  return (
    <BrowserRouter>
      <Navbar />

      <Routes>
        <Route path="/" element={<MarvelScreen />} />
        <Route path="/dc" element={<DCScreen />} />
        <Route path="/marvel" element={<MarvelScreen />} />
        <Route path="/search" element={<SearchScreen />} />
        <Route path="/login" element={<LoginScreen />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter
