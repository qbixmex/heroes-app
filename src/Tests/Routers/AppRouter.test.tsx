import { mount } from 'enzyme';
import { AuthContext } from '../../Auth/authContext';
import AppRouter from '../../Routers/AppRouter';

//@ts-ignore
describe("Test on <AppRouter />", () => {

  test("Should show login screen if user is not authenticated", () => {
    const contextValue = {
      user: {
        logged: false
      },
      dispatch: jest.fn()
    };

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find("h1").text().trim() ).toBe( "Login Screen" );
  });
  
  test("Should show marvel component screen if user is authenticated", () => {

    const contextValue = {
      user: {
        logged: true,
        name: "John Doe"
      },
      dispatch: jest.fn()
    };

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find(".navbar").exists() ).toBe( true );
  });
});
