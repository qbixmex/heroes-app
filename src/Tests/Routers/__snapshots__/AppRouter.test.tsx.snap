// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <AppRouter /> Should show login screen if user is not authenticated 1`] = `
Array [
  <h1
    className="text-center display-1 green mt-4"
  >
    Login Screen
  </h1>,
  <hr />,
  <div
    className="text-center"
  >
    <button
      className="btn btn-outline-primary mt-5"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>,
]
`;

exports[`Test on <AppRouter /> Should show marvel component screen if user is authenticated 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark bg-gradient px-4 mb-4"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <button
            className="nav-link btn text-info"
          >
            John Doe
          </button>
        </li>
        <li
          className="nav-item"
        >
          <button
            className="nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </li>
      </ul>
    </div>
  </nav>,
  <section
    className="container"
  >
    Array [
      <h1
        className="text-center display-1 green"
        id="marvelTitle"
      >
        Marvel Screen
      </h1>,
      <hr />,
      <div
        className="row rows-cols-1 animate__animated animate__fadeIn"
      >
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-spider"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Spider Man"
                    className="img-fluid rounded"
                    src="/assets/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Spider Man
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-captain"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Captain America"
                    className="img-fluid rounded"
                    src="/assets/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Captain America
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-iron"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Iron Man"
                    className="img-fluid rounded"
                    src="/assets/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Iron Man
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-thor"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Thor"
                    className="img-fluid rounded"
                    src="/assets/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Thor
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-hulk"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Hulk"
                    className="img-fluid rounded"
                    src="/assets/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Hulk
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-wolverine"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Wolverine"
                    className="img-fluid rounded"
                    src="/assets/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Wolverine
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-daredevil"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Daredevil"
                    className="img-fluid rounded"
                    src="/assets/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Daredevil
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-hawkeye"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Hawkeye"
                    className="img-fluid rounded"
                    src="/assets/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Hawkeye
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-cyclops"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Cyclops"
                    className="img-fluid rounded"
                    src="/assets/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Cyclops
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-12 col-md-6 col-lg-4 col-xl-3 animate__animated animate__fadeIn"
        >
          <a
            className="text-decoration-none"
            href="/hero/marvel-silver"
            onClick={[Function]}
            title="More Details"
          >
            <div
              className="card text-dark mb-4"
            >
              <div
                className="row no-gutters"
              >
                <div
                  className="col-12"
                >
                  <img
                    alt="Silver Surfer"
                    className="img-fluid rounded"
                    src="/assets/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-12"
                >
                  <div
                    className="card-body"
                  >
                    <h1
                      className="text-center dark"
                    >
                      Silver Surfer
                    </h1>
                  </div>
                </div>
              </div>
            </div>
          </a>
        </div>
      </div>,
    ]
  </section>,
]
`;
